dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/TCPUtils.java

```diff
@@ -0,0 +1,124 @@
+                bindFuture.channel().attr(ConfigConstants.NameKey).set(channelName);
+                bindFuture.channel().attr(ConfigConstants.HostKey).set(inetHost);
+                bindFuture.channel().attr(ConfigConstants.PortKey).set(inetPort);
+
+                Tr.info(tc, TCPMessageConstants.TCP_CHANNEL_STARTED, new Object[] { channelName, inetHost, String.valueOf(inetPort) });
+            } else {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "bindHelper failed due to: " + future.cause().getMessage());
+                }
+
+                if (retryCount > 0) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc, "attempt to bind again after a wait of " +timeBetweenRetriesMsec +"ms; " + retryCount + " attempts remaining");
+                    }
+                    // recurse until we either complete successfully or run out of retries;
+                    try {
+                        Thread.sleep(timeBetweenRetriesMsec);
+                    } catch (InterruptedException x) {
+                        // do nothing but debug
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            Tr.debug(tc, "sleep caught InterruptedException.  will proceed.");
+                        }
+                    }
+                    bindHelper(framework, bootstrap, inetHost, inetPort, bindListener, retryCount - 1);
+                } else {
+                    Tr.error(tc, TCPMessageConstants.BIND_ERROR,
+                            new Object[] { channelName, inetHost, String.valueOf(inetPort), 
+                                    bindFuture.cause().getMessage() });
+                }
+            }
+        });
+        return bindFuture;
+    }
+
+    public static ChannelFuture start(NettyFrameworkImpl framework, ServerBootstrapExtended bootstrap, String inetHost, int inetPort, ChannelFutureListener bindListener) throws NettyException {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "start (TCP): attempt to bind a channel at host " + inetHost + " port " + inetPort);
+        }
+        try {
+            framework.runWhenServerStarted(new Callable<ChannelFuture>() {
+                @Override
+                public ChannelFuture call() {
+                    TCPConfigurationImpl config = (TCPConfigurationImpl) bootstrap.getConfiguration();
+                    int bindRetryCount = config.getPortOpenRetries();
+                    return bindHelper(framework, bootstrap, inetHost, inetPort, bindListener, bindRetryCount);
+                }
+            });
+        } catch (Exception e) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "caught exception performing late cycle server startup task: " + e);
```

Comment:
It might get good to pull out the cause and message?  Or do you want to use an overloaded toString?